VultisigApp/VultisigApp/Services/CryptoPriceService.swift

```
@@ -112,7 +112,18 @@ private extension CryptoPriceService {
             
             var rates: [Rate] = []
             for contract in contracts {
-                let poolPrice = await SuiService.getTokenUSDValue(contractAddress: contract)
+                // Try to find the coin metadata to get proper decimals
+                var decimals = 9 // Default SUI token decimals
```

if we can't figure out the decimals , I would rather we throw an error